climada/hazard/low_flow.py

```diff
@@ -203,127 +203,189 @@ def set_from_nc(self, input_dir=None, centroids=None, countries=[], reg=None,
+            centroids (Centroids)"""
         # intensity = list()
 
-        uni_ev = np.unique(self.data['cluster_id'].values)
-        num_ev = uni_ev.size
+        uniq_ev = np.unique(self.data_df['cluster_id'].values)
         num_centr = centroids.size
         res_centr = self._centroids_resolution(centroids)
 
-        # For one event, if more than one points of data dataframe have the
-        # same coordinates, take the sum of days below threshold
-        # of these points (duration as accumulated intensity).
-        tree_centr = BallTree(centroids.coord, metric='chebyshev')
-        if self.pool:
-            chunksize = min(num_ev // self.pool.ncpus, 1000)
-            intensity_list = self.pool.map(self._intensity_one_cluster,
-                                           itertools.repeat(self.data, num_ev),
-                                           itertools.repeat(tree_centr, num_ev),
-                                           uni_ev, itertools.repeat(res_centr),
-                                           itertools.repeat(num_centr),
-                                           chunksize=chunksize)
-        else:
-            intensity_list = []
-            for cl_id in uni_ev:
-                intensity_list.append(
-                    self._intensity_one_cluster(self.data, tree_centr, cl_id,
-                                                res_centr, num_centr))
         self.tag = TagHazard(HAZ_TYPE)
         self.units = 'days'  # days below threshold
         self.centroids = centroids
 
         # Following values are defined for each event
-        self.event_id = np.sort(self.data.cluster_id.unique())
+        self.event_id = np.sort(self.data_df.cluster_id.unique())
         self.event_id = self.event_id[self.event_id > 0]
         self.event_name = list(map(str, self.event_id))
-        self.date = np.zeros(num_ev, int)
-        self.date_end = np.zeros(num_ev, int)
 
-        for ev_idx, ev_id in enumerate(uni_ev):
-            self.date[ev_idx] = self.data[self.data.cluster_id == ev_id].dtime.min()
-            self.date_end[ev_idx] = self.data[self.data.cluster_id == ev_id].dtime.max()
-        self.orig = np.ones(num_ev, bool)
+        self._set_dates(uniq_ev)
+
+        self.orig = np.ones(uniq_ev.size)
         self.set_frequency()
 
+        self.intensity = self._intensity_loop(uniq_ev, centroids.coord, res_centr, num_centr)
+
         # Following values are defined for each event and centroid
-        self.intensity = sparse.lil_matrix(np.zeros((num_ev, num_centr)))
-        for idx, ev_intensity in enumerate(intensity_list):
-            self.intensity[idx] = ev_intensity
         self.intensity = self.intensity.tocsr()
         self.fraction = self.intensity.copy()
         self.fraction.data.fill(1.0)
 
     def identify_clusters(self, clus_thres_xy=None, clus_thresh_t=None, min_samples=None):
-        """call clustering and set events"""
+        """call clustering functions and set events in hazard
```

Comment:
As I understand you are not "setting events in hazard" but only identify the clusters inside the dataframe